src/ui/index.js

```diff
@@ -83,12 +83,12 @@ export default async (kbnServer, server, config) => {
       vars: await reduceAsync(
         uiExports.injectedVarsReplacers,
         async (acc, replacer) => await replacer(acc, request, server),
-        defaults(await app.getInjectedVars() || {}, uiExports.defaultInjectedVars)
+        defaults(await app.getInjectedVars() || {}, uiExports.defaultInjectedVars, vars)
```

Comment:
I think we should rename `vars` -> `injectedVarsOverrides` and move it to the front of the list here. If the code calling `renderApp()` has specific vars in mind they shouldn't be overridden by the defaults or the standard injected vars, only the "replacers"  could override them.